openeo_processes_dask/process_implementations/comparison.py

```diff
@@ -0,0 +1,140 @@
+from typing import Callable, Optional
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+from numpy.typing import ArrayLike
+from xarray.core.duck_array_ops import notnull
+
+__all__ = [
+    "is_infinite",
+    "is_valid",
+    "is_nodata",
+    "is_nan",
+    "eq",
+    "neq",
+    "gt",
+    "gte",
+    "lt",
+    "lte",
+    "between",
+]
+
+
+def is_infinite(x: ArrayLike):
+    if x is None:
+        return False
+    if (
+        type(x) in [str, list, dict]
+        or type(x) in [np.ndarray, da.core.Array]
+        and x.dtype.kind.lower() in ["u", "s", "o"]
+    ):
+        return False
+    return np.isinf(x)
+
+
+def is_valid(x: ArrayLike):
+    finite = np.logical_not(is_infinite(x))
+    return np.logical_and(notnull(x), finite)
+
+
+def is_nodata(x: ArrayLike):
+    return np.logical_not(is_valid(x))
+
+
+def is_nan(x: ArrayLike):
+    return is_nodata(x)
+
+
+def eq(
+    x: ArrayLike,
+    y: ArrayLike,
+    delta: Optional[float] = None,
+    case_sensitive: Optional[bool] = True,
+):
+    if not type(x) in [np.ndarray, da.core.Array] and not type(y) in [
+        np.ndarray,
+        da.core.Array,
+    ]:
+        if is_nodata(x) or is_nodata(y):
+            return np.nan
+    if x is False or y is False:
+        return False
+    if delta:
+        ar_eq = np.isclose(x, y, atol=delta)
+    elif not case_sensitive:
+        ar_eq = np.char.lower(x) == np.char.lower(y)
+    else:
+        ar_eq = x == y
+    return np.where(np.logical_and(is_valid(x), is_valid(y)), ar_eq, np.nan)
+
+
+def neq(
+    x: ArrayLike,
+    y: ArrayLike,
+    delta: Optional[float] = None,
+    case_sensitive: Optional[bool] = True,
+):
+    eq_val = eq(x, y, delta=delta, case_sensitive=case_sensitive)
+    return np.where(
+        np.logical_and(is_valid(x), is_valid(y)), np.logical_not(eq_val), np.nan
```

Comment:
This applies for all the processes, but I think this shouldn't use `is_valid` (which also considers infinity to be invalid), but just the `notnull` function

Reply:
So, the question is if we want to consider infinity as a valid or invalid value, right? I am not sure if we want to compare two infinite values with each other, e.g.  np.inf == np.inf -> True. Do we want this behaviour?

Reply:
Hmm, I think there's two questions here:
1) Is infinity a valid number in `is_valid`?
-> No, according to `is_valid` it isn't
2) is infinity null?
-> Also no, it's not nodata

I think the comparison processes just want to use `notnull` to check for nullness, rather than the `is_valid` process.

I'm not 100% sure on this, but since infinity is represented as a constant in IEEE 754 (see https://numpy.org/devdocs/reference/constants.html#numpy.Inf), I think `np.inf == np.inf -> True` seems right to me?

Reply:
okay, I now did some additional tests with numpy and inf comparison, and I think we can stick to how numpy handles this. I will push a new version asap :)